cpp/source/deal.II/geometry.templates.h

```diff
@@ -14,165 +14,37 @@
-  anode_collector_bbox =
-      std::make_pair(dealii::Point<3>(-anode_collector_width, 0.0, 0.0),
-                     dealii::Point<3>(0.0, sandwich_height, sandwich_depth));
-  anode_electrode_bbox = std::make_pair(
-      dealii::Point<3>(0.0, 0.0, 0.0),
-      dealii::Point<3>(anode_electrode_width, sandwich_height, sandwich_depth));
-  separator_bbox =
-      std::make_pair(dealii::Point<3>(anode_electrode_width, 0.0, 0.0),
-                     dealii::Point<3>(anode_electrode_width + separator_width,
-                                      sandwich_height, sandwich_depth));
-  cathode_electrode_bbox = std::make_pair(
-      dealii::Point<3>(anode_electrode_width + separator_width, 0.0, 0.0),
-      dealii::Point<3>(anode_electrode_width + separator_width +
-                           cathode_electrode_width,
-                       sandwich_height, sandwich_depth));
-  cathode_collector_bbox = std::make_pair(
-      dealii::Point<3>(anode_electrode_width + separator_width +
-                           cathode_electrode_width,
-                       0.0, 0.0),
-      dealii::Point<3>(anode_electrode_width + separator_width +
-                           cathode_electrode_width + cathode_collector_width,
-                       sandwich_height, sandwich_depth));
-  cathode_tab_bbox = std::make_pair(
-      dealii::Point<3>(anode_electrode_width + separator_width +
-                           cathode_electrode_width,
-                       -tab_height, 0.0),
-      dealii::Point<3>(anode_electrode_width + separator_width +
-                           cathode_electrode_width + cathode_collector_width,
-                       0.0, sandwich_depth));
-  return std::make_tuple(anode_tab_bbox, anode_collector_bbox,
-                         anode_electrode_bbox, separator_bbox,
-                         cathode_electrode_bbox, cathode_collector_bbox,
-                         cathode_tab_bbox);
+  materials = std::make_shared<std::unordered_map<
+      std::string, std::vector<dealii::types::material_id>>>();
+  int const n_materials = database->get<int>("materials");
+  for (int m = 0; m < n_materials; ++m)
+  {
+    boost::property_tree::ptree material_database =
+        database->get_child("material_" + std::to_string(m));
+    std::vector<dealii::types::material_id> const material_ids =
+        to_vector<dealii::types::material_id>(
+            material_database.get<std::string>("material_id"));
+    std::string const material_name =
+        material_database.get<std::string>("name");
+    materials->emplace(material_name, material_ids);
+  }
 }
 
 template <int dim>
 Geometry<dim>::Geometry(
-    std::shared_ptr<boost::property_tree::ptree const> const &database)
+    std::shared_ptr<boost::property_tree::ptree const> database,
+    boost::mpi::communicator mpi_communicator)
+    : mpi_communicator(mpi_communicator)
 {
-  this->triangulation = std::make_shared<dealii::Triangulation<dim>>();
```

Comment:
I agree that ``this->`` everywhere to signify data member is a bit heavy but I still like in general being able to tell.
How about prefixing it with underscores ``_triangulation``? What do you think?